README.md

```diff
@@ -74,6 +74,108 @@ This has been tested on Ubuntu 20.04 but the same logic should apply to any syst
+
+Debugging is also supported in VS Code. Here is an example launch and task files necessary. Some changes to properly resolve paths are needed (changes to the user and version might be required):
+launch.json
+````json
+    {
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "OpenOCD Debug",
+            "type": "cppdbg",
+            "request": "launch",
+            "miDebuggerPath": "/home/casto/.espressif/tools/xtensa-esp32-elf/esp-2020r2-8.2.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gdb", // to be changed by user
+            "program": "${workspaceFolder}/cmake-build-target/esp32-boilerplate.elf",
+            "preLaunchTask": "openocd",
+            "setupCommands": [{
+                "description": "enable pretty printing for gdb",
+                "text": "-enable-pretty-printing",
+                "ignoreFailures": true
+            }, {
+                "text": "file '${workspaceFolder}/cmake-build-target/esp32-boilerplate.elf'"
+            }, {
+                "text": "target remote :3333"
+            }, {
+                "text": "monitor program_esp32 ${workspaceFolder}/cmake-build-target/esp32-boilerplate.elf 0x10000 verify"
+            }, {
+                "text": "monitor reset halt"
+            }, {
+                "text": "thb app_main"
+            }],
+            "cwd": "${workspaceFolder}",
+            "externalConsole": false
+        }
+    ]
+}
+````
+tasks.json
+````json
+    {
+    "version": "2.0.0",
+    "tasks": [
+        {
+            "label": "openocd",
+            "type": "shell",
+            "isBackground": true,
+            "problemMatcher": [
+                {
+                  "pattern": [
+                    {
+                      "regexp": ".",
+                      "file": 1,
+                      "location": 2,
+                      "message": 3
+                    }
+                  ],
+                  "background": {
+                    "activeOnStart": true,
+                    "beginsPattern": ".",
+                    "endsPattern": ".",
+                  }
+                }
+              ],
+            "options": 
+            {
+                "cwd": "/home/casto/.espressif/tools/openocd-esp32/v0.10.0-esp32-20191114/openocd-esp32" // to be changed by user
+            },
+            "command": "bin/openocd -s share/openocd/scripts -f ${workspaceFolder}/tools/openocd/adafruit-esp.cfg -c 'program_esp ${workspaceFolder}/cmake-build-target/esp32-boilerplate.bin 0x10000'",
```

Comment:
woulnd't it be more convenient to have this in the launch.json. The field openocd would be as the debugServerPath and use the debugServerArgs for the args. 

Then have a task.json as preLaunchTask that will build your project (so it calls cd $dir && make)  so your build is up to date?

Reply:
I had trouble getting the debugServer with vs code and documentation is hard to find on this type of configuration. I added a make command to the preLaunchTask so the build is up to date